### DISCUSSION TITLE
Help needed with forge fmt

### DISCUSSION THREAD

**Discussion by gaavar** [foundry-rs/foundry#3889] [Help]
Hello! I am working on migrating a project from prettier to fmt.

I want to make sure that the function parameters are kept multi-line instead of being put together in the same line.

The `fmt` tool by default just puts the function parameters in the same line.

<img width="714" alt="image" src="https://user-images.githubusercontent.com/33789105/207449080-aa87cce2-9c68-4a20-a9c8-3ed7c35228c9.png">

I have tried exploring `multiline_func_header` and `single_line_block` but neither of them work.

Can someone please help me out with this?

**Comment by KholdStare** [comment]
Have you tried checking the `line_length` setting? https://book.getfoundry.sh/reference/config/formatter#line_length

prettier and fmt have different defaults.

**Reply by gaavar** [reply]
Thank you for your response! Although adjusting `line_length` might achieve that purpose, it will format all lines in a certain way which is undesirable.

There's been a lot of similar discussions in GitHub issue posts regarding `multiline_func_header` that was developed I believe for cases similar to mine. I was unable to get some traction from the team for a response :/
